src/redux/sagas/transactions/transactionsToDb.ts

```diff
@@ -36,7 +36,7 @@ const handleBeforeUnload = (e) => {
   return e.returnValue;
 };
 
-const onTransactionPending = (id: string) => {
```

Comment:
Honestly, I'm not sure if we even need this anymore, could you check?

Reply:
Okay I've looked into this and seems like it is worth keeping around. It is intended to prevent you from accidentally closing the tab whilst a transaction is pending:

<img width="303" alt="Screenshot 2024-08-27 at 09 30 38" src="https://github.com/user-attachments/assets/6ad5ba72-ddd0-407e-9063-bc4ad9d1b10f">

Although I think a better implementation of this might be to add the `beforeunload` event listener at the very start of the saga and remove it at the end, to prevent accidentally closing the tab at any stage of the saga (rather than just around the transactions at it is currently). Either way this would be better tackled as it's own issue.